### PR TITLE
Change git current branch command to keep compatibility with git 2.20.1

### PR DESCRIPTION
**Pull request by felipecrescencio-cit**
Fixes an error in step `5-app-infra` applying terraform changes using Cloud Build.
As Cloud Build is based on step `4-projects` docker image having git version 2.20.1 the command:
`git branch --show-current`
does not work because it requires Git 2.22.0+.
To keep compatibility with git version 2.20.1 the command was changed.

### CODE REVIEW

**Review by bharathkkb — COMMENTED** [review]
Let's also update https://github.com/terraform-google-modules/terraform-example-foundation/blob/97d31cf6ae1878f59875f17d56ab5988aa74de96/scripts/validate-requirements.sh#L24 so we have the same common min version

**Comment by felipecrescencio-cit** [comment]
> Let's also update
> 
> https://github.com/terraform-google-modules/terraform-example-foundation/blob/97d31cf6ae1878f59875f17d56ab5988aa74de96/scripts/validate-requirements.sh#L24
> 
> so we have the same common min version

Done

**Comment by felipecrescencio-cit** [comment]
> Let's also update
> 
> https://github.com/terraform-google-modules/terraform-example-foundation/blob/97d31cf6ae1878f59875f17d56ab5988aa74de96/scripts/validate-requirements.sh#L24
> 
> so we have the same common min version

@bharathkkb I had to revert git min version to 2.28.0 again.
This version is needed because of `git config --global init.defaultBranch main` command that requires git 2.28.0

**Comment by felipecrescencio-cit** [comment]
> Let's also update
> 
> https://github.com/terraform-google-modules/terraform-example-foundation/blob/97d31cf6ae1878f59875f17d56ab5988aa74de96/scripts/validate-requirements.sh#L24
> 
> so we have the same common min version

**Review by felipecrescencio-cit — COMMENTED** [review]
@bharathkkb 
I opened issue #812 to solve the root cause of this compatibility command.

File: build/tf-wrapper.sh [hunk 1]
```diff
@@ -140,7 +140,11 @@ tf_validate() {
           gcloud source repos clone gcp-policies "${policy_file_path}" --project="${project_id}" || exit 34
           pushd .
           cd "${policy_file_path}"
-          current_branch=$(git branch --show-current)
+          # Commented command below works only on Git 2.22.0+
+          # current_branch=$(git branch --show-current)
+          # As Cloud Build is based on step 4-projects docker image having
+          # git version 2.20.1 installed the command below keeps compatibility
+          current_branch=$(git symbolic-ref --short HEAD)
```
Comment (felipecrescencio-cit) [inline build/tf-wrapper.sh:147]: @bharathkkb 
I opened issue #812 to solve the root cause of this compatibility command.

**Review by bharathkkb — APPROVED** [review]

**Pull request merged**